istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,714 @@
+  --injection --injection-config istio-sidecar-injector
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				return fmt.Errorf("err when creating Kubernetes client interface: %v", err)
+			}
+			validationErr, injectionErr := disableWebhookConfig(client, opts)
+			if validationErr != nil || injectionErr != nil {
+				return fmt.Errorf("error when disabling webhook configurations. validation err: %v. injection err: %v",
+					validationErr, injectionErr)
+			}
+			fmt.Println("webhook configurations have been disabled")
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.disableValidationWebhook, "validation", true, "Disable "+
+		"validating webhook (default true).")
+	flags.StringVar(&opts.validatingWebhookConfigName, "validation-config", "istio-galley",
+		"The validating webhook configuration to disable.")
+	flags.BoolVar(&opts.disableInjectionWebhook, "injection", true, "Disable "+
+		"mutating webhook (default true).")
+	flags.StringVar(&opts.mutatingWebhookConfigName, "injection-config", "istio-sidecar-injector",
+		"The mutating webhook configuration to disable.")
+
+	return cmd
+}
+
+func newStatusCmd() *cobra.Command {
+	opts := &statusCliOptions{}
+	cmd := &cobra.Command{
+		Use:   "status",
+		Short: "Get webhook configurations",
+		Example: `
+# Display the webhook configuration of Galley
+istioctl experimental post-install webhook status --validation --validation-config istio-galley
+# Display the webhook configuration of Galley and Sidecar Injector
+istioctl experimental post-install webhook status --validation --validation-config istio-galley 
+  --injection --injection-config istio-sidecar-injector
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				return fmt.Errorf("err when creating Kubernetes client interface: %v", err)
+			}
+			validationErr, injectionErr := displayWebhookConfig(client, opts)
+			if validationErr != nil || injectionErr != nil {
+				return fmt.Errorf("error when displaying webhook configurations. validation err: %v. injection err: %v",
+					validationErr, injectionErr)
+			}
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.validationWebhook, "validation", true, "Display "+
+		"the validating webhook configuration.")
+	flags.BoolVar(&opts.injectionWebhook, "injection", true, "Display "+
+		"the injection webhook configuration.")
+	flags.StringVar(&opts.validatingWebhookConfigName, "validation-config", "istio-galley",
+		"The name of the ValidatingWebhookConfiguration to display.")
+	flags.StringVar(&opts.mutatingWebhookConfigName, "injection-config", "istio-sidecar-injector",
+		"The name of the MutatingWebhookConfiguration to display.")
```

Comment:
The validation and injection k8s webhook resource name is encoded in the provided file(s). I don't see much value in allowing the user to overwrite it through the command line. They would still need to customize the YAML to avoid duplicate webhook names (e.g. injection.istio.io).

Reply:
This command display the status of a webhook config; its input parameters do not include a config file and which status to display is based on the webhook config name.

Reply:
Ah, right. However these could still be removed from enable and disable commands since they have the config path input.

Reply:
Done: removed the config name from the enable command. For disable command, as it does not take config file as a parameter, the config name is still needed.